### ISSUE TITLE
False positive on 'await await[0]' in top level

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#790]
```js
let await = [Promise.resolve('hello')];
console.log(await await[0]);
```
quick-lint-js incorrectly reports redundant 'await' [E0266] for the above code. However, there is no redundant await.
